utf8/reader.go

```diff
@@ -0,0 +1,142 @@
+package utf8
+
+import (
+	"fmt"
+	"io"
+	"unicode/utf8"
+
+	"github.com/madlambda/spells/io/bytes"
+)
+
+type (
+	RuneReader struct {
+		r io.RuneReader
+	}
+
+	// ReaderReader implements a runes.Reader from a bytes.Reader interface.
+	ReaderReader struct {
```

Comment:
Now we can have just utf8.Reader no ? 🚀 

```suggestion
	Reader struct {
```

Or, started thinking on these terms:

```suggestion
	Decoder struct {
```